src/js/BookReader.js

```diff
@@ -2149,99 +2151,67 @@ BookReader.prototype.createPopup = createPopup;
  * Bind navigation handlers
  */
 BookReader.prototype.bindNavigationHandlers = function() {
-  var self = this;
+  const self = this;
 
   // Note the mobile plugin attaches itself to body, so we need to select outside
-  var jIcons = this.$('.BRicon').add('.BRmobileMenu .BRicon');
-
-  jIcons.filter('.onepg').bind('click', function() {
-    self.switchMode(self.constMode1up);
-  });
-
-  jIcons.filter('.twopg').bind('click', function() {
-    self.switchMode(self.constMode2up);
-  });
-
-  jIcons.filter('.thumb').bind('click', function() {
-    self.switchMode(self.constModeThumb);
-  });
+  const jIcons = this.$('.BRicon').add('.BRmobileMenu .BRicon');
+  const navigationControls = {
+    book_left: () => {
+      this.trigger(BookReader.eventNames.stop);
+      this.left();
+    },
+    book_right: () => {
+      this.trigger(BookReader.eventNames.stop);
+      this.right();
+    },
+    book_up: () => {
+      if ($.inArray(this.mode, [this.constMode1up, this.constModeThumb]) >= 0) {
+        this.scrollUp();
+      } else {
+        this.prev();
+      }
+    },
+    book_down: () => {
+      if ($.inArray(this.mode, [this.constMode1up, this.constModeThumb]) >= 0) {
+        this.scrollDown();
+      } else {
+        this.next();
+      }
+    },
+    book_top: this.first,
+    book_bottom: this.last,
+    book_leftmost: this.leftmost,
+    book_rightmost: this.rightmost,
+    zoom_in: () => {
+      this.trigger(BookReader.eventNames.stop);
+      this.zoom(1);
+      this.trigger(BookReader.eventNames.zoomIn);
+    },
+    zoom_out: () => {
+      this.trigger(BookReader.eventNames.stop);
+      this.zoom(-1);
+      this.trigger(BookReader.eventNames.zoomOut);
+    },
+    full: () => {
+      if (this.ui == 'embed') {
+        var url = this.$('.BRembedreturn a').attr('href');
+        window.open(url);
+      } else {
+        this.toggleFullscreen();
+      }
+    },
+  };
 
   jIcons.filter('.fit').bind('fit', function() {
     // XXXmang implement autofit zoom
   });
```

Comment:
Let's copy into the new object as is (with the comment) for consistency :+1:

Reply:
All of the events on the `navigationControls` object are bound to a click event. This one was a custom 'fit' event, so I left it on its own.